### ISSUE TITLE
High CPU usage of Pilot due to ServiceEntry prior to 1.5.0

### ISSUE THREAD

**Issue by cagataygurturk** [istio/istio#20647]
**Bug description**

We have a namespace which is running Pods without sidecars, so this namespace is outside of the mesh. Our mesh is set to `REGISTRY_ONLY`, therefore we are creating the following `ServiceEntry` in order to let Pods in the mesh to connect to those outside of the mesh:

```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: elasticsearch
  namespace: istio-system
spec:
  hosts:
    - elasticsearch-saved-searches-client.nonprod-buyer-ca-els.svc.cluster.local
    - elasticsearch-ad-search-client.nonprod-buyer-ca-els.svc.cluster.local
  location: MESH_EXTERNAL
  ports:
    - number: 9200
      name: http
      protocol: HTTP
  resolution: DNS
```
Prior to 1.5.0-alpha.0, whenever we create this object, Pilot starts to use up to 100% CPU, its HPA kicks in and new Pilot replicas are created. Interestingly, also the new Pilot replicas uses the CPU up to 100%. This disrupts the whole mesh and there is also no log indicating a problem.

When we replace the hostname with a not existing service, CPU usage goes down to normal:

```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: elasticsearch
  namespace: istio-system
spec:
  hosts:
    - test-not-existing-service.nonprod-buyer-ca-els.svc.cluster.local
  location: MESH_EXTERNAL
  ports:
    - number: 9200
      name: http
      protocol: HTTP
  resolution: DNS
```

With 1.5.0.alpha.0 everything works as expected.

If this was a known bug and it was fixed in 1.5.0, I would be happy to know the issue/PR.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ x] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Comment by howardjohn** [comment]
likely fixed by changes @ramaraochavali made

**Comment by ramaraochavali** [comment]
There were multiple PRs to optimize the service entry behaviour in 1.5-alpha. 
https://github.com/istio/istio/pull/19305
https://github.com/istio/istio/pull/19936

There are few more actually. Are you looking for any specific change?

**Comment by cagataygurturk** [comment]
@ramaraochavali No, I am not looking for any specific chance but I just wondered whether if this was an known issue.

Any trick to make this `ServiceEntry` work in 1.4.3 anyways, or will these fixes be included in 1.4.4?

@howardjohn please decide if you want to close this issue.

**Comment by 2ZZ** [comment]
Hi, I am unable to test on 1.5.0 but I observed a similar issue when a ServiceEntry containing a low TTL DNS name was deployed to a 36 node cluster with ~1000 pods.

```
---
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: a-service-entry
spec:
  hosts:
  - <dns-record-with-15-second-ttl-here>
  location: MESH_EXTERNAL
  ports:
  - name: http
    number: 443
    protocol: HTTPS
  resolution: DNS
```

When created Pilot CPU goes up dramatically during deployments and the Envoy sidecar starts to take a long time to get its config.

```
2020-02-20T15:16:41.753458Z     info    Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?): cds updates: 2 successful, 0 rejected; lds updates: 0 successful, 0 rejected
...
<snip>
...
2020-02-20T15:19:19.753972Z     info    Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?): cds updates: 17 successful, 0 rejected; lds updates: 0 successful, 0 rejected
```

**Comment by howardjohn** [comment]
That is unlikely to be related to this issue can you open a new issue with more info? the ttl should have no impact on pilot at all, envoy does the resolution so I am surprised it has any impact at all

**Comment by 2ZZ** [comment]
Hi @howardjohn, you are right, I retested and TTL is not the cause. I have created a new issue https://github.com/istio/istio/issues/21348

**Comment by howardjohn** [comment]
Given we are not planning to backport the changes, and the fixes are in 1.5+ which has been out for a while, I think it makes sense to close this. thanks!
